test/distributed/test_c10d_nccl.py

```diff
@@ -2554,6 +2554,24 @@ def _test_nccl_errors_blocking(self, func):
             del process_group
             func()
 
+    def _test_barrier_error(self):
```

Comment:
nit: if this method do not have more than 1 call site, we can consider flatten it.

Reply:
put it as a private method as it is shared by both  test_nccl_blocking_wait_with_barrier and test_nccl_nonblocking_wait_with_barrier